tools/BuildPackagesTask/VerifyAuthenticodeSignatureTask.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace Microsoft.Azure.Build.Tasks
+{
+    using Microsoft.Build.Framework;
+    using Microsoft.Build.Utilities;
+    using System;
+    using System.IO;
+    using System.Security.Cryptography.X509Certificates;
+
+    public class VerifyAuthenticodeSignatureTask : Task
+    {
+        /// <summary>
+        /// Gets or sets the assembly on which authenticode signature verification is performed.
+        /// </summary>
+        [Required]
+        public ITaskItem Assembly { get; set; }
+
+        [Output]
+        public bool AuthTaskErrorsDetected { get; private set; }
+
+        public override bool Execute()
+        {
+            string assemblyPath = Assembly.ItemSpec;
+            string assemblyName = Path.GetFileName(assemblyPath);            
+            AuthTaskErrorsDetected = false;
+            try
+            {
+                Log.LogMessage("Verifying Authenticode Signature for '{0}' - '{1}'", assemblyName, assemblyPath);
+                X509Certificate cert = X509Certificate.CreateFromSignedFile(assemblyPath);
+                if ((assemblyName.ToLower().StartsWith("microsoft")) || (assemblyName.ToLower().StartsWith("system")))
+                {
+                    if (cert.Issuer.ToLower().Contains("microsoft"))
+                    {
+                        // Verify if the certificate by which it was signed is not expired
+                        DateTime notAfter = DateTime.Parse(cert.GetExpirationDateString());
```

Comment:
I think we need to do a full valididty check here, including crl.  Are there simple APIs we can use to detect that the cert is valid.